examples/js/effects/VREffect.js

```diff
@@ -55,31 +69,88 @@ THREE.VREffect = function ( renderer, onError ) {
 
 	// fullscreen
 
-	var isFullscreen = false;
+	var isPresenting = false;
 
 	var canvas = renderer.domElement;
 	var fullscreenchange = canvas.mozRequestFullScreen ? 'mozfullscreenchange' : 'webkitfullscreenchange';
 
-	document.addEventListener( fullscreenchange, function ( event ) {
+	document.addEventListener( fullscreenchange, function () {
+
+		if ( vrHMD && deprecatedAPI ) {
+
+			isPresenting = document.mozFullScreenElement || document.webkitFullscreenElement;
+
+		}
+
+	}, false );
+
+	window.addEventListener( 'vrdisplaypresentchange', function () {
 
-		isFullscreen = document.mozFullScreenElement || document.webkitFullscreenElement;
+		isPresenting = vrHMD && vrHMD.isPresenting;
 
 	}, false );
 
 	this.setFullScreen = function ( boolean ) {
 
-		if ( vrHMD === undefined ) return;
-		if ( isFullscreen === boolean ) return;
+		return new Promise(function(resolve, reject) {
```

Comment:
I'd prefer to return a Promise, but I noticed that @mrdoob isn't using promises anywhere else in the codebase. if you don't want to polyfill, perhaps you could do some conditional logic that returns a `Promise` only if `Promise` is defined on `window`. and then possibly define a `reject` callback that does `console.error` perhaps?